students/ArchWu/Session5/mailroom.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+import sys
+import operator
+import tempfile
+#donations[]
+#donors {donor1:donations1, donor2:donations2}
+donors = {}
+# The data struction used in this program, a list of tuples
+
+
+def add_donor(name, donation = 0):
+    """ Adds a new donor with an initial donation """
+    donations = [donation]
+    donors[name] = donations
+    return
+
+def add_donation(name, donation):
+    """ Adds a donation to an existing donor """
+    donors[name].append(donation)
+    return
+
+def report():
+    """ Make a report on donations received by a style as specified """
+    print("Donor Name                | Total Given | Num Gifts | Average Gift")
+    print("------------------------------------------------------------------")
+    sorted_donors = sorted(donors.items(), reverse = True, key = lambda item: (sum(item[1])))
+    return [print("{0:25s}   {1:11.2f}   {2:9d}   {3:12.2f}".format(*gen_stats(donor))) for donor in sorted_donors]
+
+
+def gen_stats(donor):
+    """ A function calculates the total and average donation of a donor """
+    name = donor[0]
+    donations = donor[1]
+    total = round(sum(donations),2)
+    num_gifts = len(donations)
+    avg = round((total / num_gifts), 2)
+    return (name, total, num_gifts, avg)
+
+def thank_you():
+    """ Thank-you funcion that generates an email to thank new donations """
+    while True:
+        answer = input("Please type the Full Name, type quit to go back to previous menu> ")
+        if answer == 'list':
+            for donor in donors:
+                print(donor)
+        elif answer == 'quit':
+            return
+        elif answer in [donor[0] for donor in donors]:
```

Comment:
Be careful, you don't want to do this. donor[0] for donor in donors is the first character of each donor. I don't believe this is your intended action here.

Reply:
You do not need a comprehension here. You only want to check if answer is in donors